### PR TITLE
feat: support trusted peer ids without address

### PR DESCRIPTION
**Pull request by mattsse**
ref https://github.com/paradigmxyz/reth/issues/7003

adds support for registering peerids as trusted

needs a followup to integrate in rpc

### CODE REVIEW

File: crates/net/network/src/peers/manager.rs [hunk 1]
```diff
@@ -88,6 +88,11 @@ impl PeersHandle {
 pub struct PeersManager {
     /// All peers known to the network
     peers: HashMap<PeerId, Peer>,
+    /// The set of trusted peer ids.
+    ///
+    /// This tracks peer ids that are considered trusted, but for which we don't necessarily have
+    /// an address: [Self::add_trusted_peer_id]
+    trusted_peer_ids: HashSet<PeerId>,
```
Comment (emhane) [inline crates/net/network/src/peers/manager.rs:95]: could we please put bounds on this? would be nice if we do this with all new data structures, to not create more debt

**Review by emhane — DISMISSED** [review]

File: crates/net/network/src/peers/manager.rs (hunk 1 above)
Comment (mattsse) [reply crates/net/network/src/peers/manager.rs:95]: this is privileged, bounds are not useful here imo

**Review by Rjected — APPROVED** [review]
this looks good to me

File: crates/net/network/src/peers/manager.rs (hunk 1 above)
Comment (emhane) [reply crates/net/network/src/peers/manager.rs:95]: that is relying on accurate implementation. for sake of extensibility, it's better to set the bounds that can be derived from the parent collection. if some developer in the future forgets to remove an entry, then we get a leak.

could be nice to have types `BoundedHashSet` and `BoundedHashMap` in common crate, that tracks number of entries on insertion and remove, without lru complexity, returning error if trying on insertion when its full. on the other hand, other tools for catching memory leakage as early as possible may be better.

**Pull request merged**